src/helpers/documentation.js

```diff
@@ -235,9 +209,85 @@ export const buildDocsContents = (data, rootDir) => {
     insertPageData(relPath, contents, articles, rootPageData);
   });
 
+  // Now that content is in proper order, travel through tree and add Next Up link to each article
+  const contentsList = Object.values(contents);
+  contentsList.forEach((topicData, topicIndex) => {
+    const topicArticles = Object.values(topicData.articles);
+    const nextTopic = contentsList[topicIndex + 1];
+
+    addNextUpToArticles(topicArticles, 0, nextTopic);
+  });
   return contents;
 };
 
+/**
+ * addNextUpToArticles recursively travels through contents tree to add a Next Up property to each
+ * @param {array} articles List of articles of the current topic.
+ * @param {number} articleIndex The index of the article we're working with
+ * @param {number} nextTopic List of articles of the next topic
+ * @returns {void}
+ */
+const addNextUpToArticles = (articles, articleIndex, nextTopic) => {
+  if (articles.length === articleIndex) return;
+  const articleData = articles[articleIndex];
+  if (articleData && articleData.articles) {
+    addNextUpToArticles(Object.values(articleData.articles), 0, nextTopic);
+  } else {
+    Object.assign(articleData, {
+      nextUp: nextUpLink(articles, articleIndex, nextTopic),
+    });
+    addNextUpToArticles(articles, articleIndex + 1, nextTopic);
+  }
+};
+
+/**
+ * nextUpLink returns the url and title of the next article, whether it's in the same topic or if it's in the next topic
+ * @param {array} topicArticles List of articles of the current topic.
+ * @param {number} articleIndex The index of the article we're working with
+ * @param {number} nextTopic List of articles of the next topic
+ * @returns {object} { title, url }
+ */
+export const nextUpLink = (topicArticles, articleIndex, nextTopic) => {
+  // // Go to next topic
+  if (topicArticles.length === articleIndex + 1) {
+    // We've reached the end
+    if (!nextTopic) {
+      return null;
+    }
+    const nextTopicFirstArticle = findNextFirstArticle(nextTopic);
+
+    return {
+      title: nextTopicFirstArticle.title,
+      url: nextTopicFirstArticle.url,
+    };
+  }
+
+  // Go to next article in topic
+  const nextChild = topicArticles[articleIndex + 1];
+  const nextNestedFirstArticle = findNextFirstArticle(nextChild);
+
+  return {
+    title: nextNestedFirstArticle.title,
+    url: nextNestedFirstArticle.url,
+  };
+};
+
+/**
+ * findNextFirstArticle recursively drills down nested articles to find the first article we can link to
+ * @param {object} articleData List of articles of the current topic.
+ * @returns {object} { title, url }
+ */
+const findNextFirstArticle = (articleData) => {
```

Comment:
this helper method just find the next article we can link to. in some cases, it's just the next sibling. but in others, it's a nested collection and we have to drill down to find an article